src/torchpme/tuning/base.py

```diff
@@ -0,0 +1,247 @@
+import math
+from typing import Optional
+
+import torch
+import vesin.torch
+
+
+class TunerBase:
+    def __init__(
+        self,
+        charges: torch.Tensor,
+        cell: torch.Tensor,
+        positions: torch.Tensor,
+        cutoff: float,
+        calculator,
+        params: list[dict],
+        exponent: int = 1,
+        neighbor_indices: Optional[torch.Tensor] = None,
+        neighbor_distances: Optional[torch.Tensor] = None,
+    ):
+        self._validate_parameters(charges, cell, positions, exponent)
+        self.charges = charges
+        self.cell = cell
+        self.positions = positions
+        self.cutoff = cutoff
+        self.exponent = exponent
+        self._dtype = cell.dtype
+        self._device = cell.device
+
```

Comment:
Should we put the dtype here as argument or deduce it from the claculator.

What do you say @E-Rum ?